plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ReusableConnectionFactory.java

```diff
@@ -174,11 +179,39 @@ public void close()
             }
             closed = true;
             if (dirty) {
```

Comment:
can we simply check if the connection that is returned to the cache here by this `close()` method is not closed, and if, just not to put it into `connections`?